includes/class-admin-settings.php

```diff
@@ -0,0 +1,516 @@
+
+	/**
+	 * The Constructor.
+	 */
+	public function __construct() {
+		add_action( 'admin_menu', array( $this, 'register_admin_menu' ) );
+		add_action( 'admin_init', array( $this, 'reset_settings' ) );
+		add_action( 'admin_init', array( $this, 'register_settings' ) );
+		add_action( 'admin_notices', array( $this, 'reset_admin_notice' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ) );
+	}
+
+	/**
+	 * Initialize Class.
+	 *
+	 * @return object
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Get the default values for the settings.
+	 *
+	 * @return array The default values.
+	 */
+	private function get_default_settings() {
+		$options              = array();
+		$options['api_hosts'] = array( 'api.aspirecloud.org' );
+		return $options;
+	}
+
+	/**
+	 * Handles the Reset Functionality and triggers a Notice to inform the same.
+	 *
+	 * @return void
+	 */
+	public function reset_settings() {
+		if (
+			isset( $_GET['reset'] ) &&
+			( 'reset' === $_GET['reset'] ) &&
+			isset( $_GET['reset-nonce'] ) &&
+			wp_verify_nonce( sanitize_key( $_GET['reset-nonce'] ), 'aspirepress-reset-nonce' )
+		) {
+			$options = $this->get_default_settings();
+			update_option( $this->option_name, $options );
+			update_option( 'aspirepress-reset', 'true' );
+
+			wp_safe_redirect(
+				add_query_arg(
+					array(
+						'reset-success'       => 'success',
+						'reset-success-nonce' => wp_create_nonce( 'aspirepress-reset-success-nonce' ),
+					),
+					admin_url( 'options-general.php?page=aspirepress-settings' )
+				)
+			);
+			exit;
+		}
+	}
+
+	/**
+	 * The Admin Notice to convey a Reset Operation has happened.
+	 *
+	 * @return void
+	 */
+	public function reset_admin_notice() {
+		if (
+			( 'true' === get_option( 'aspirepress-reset' ) ) &&
+			isset( $_GET['reset-success'] ) &&
+			( 'success' === $_GET['reset-success'] ) &&
+			isset( $_GET['reset-success-nonce'] ) &&
+			wp_verify_nonce( sanitize_key( $_GET['reset-success-nonce'] ), 'aspirepress-reset-success-nonce' )
+		) {
+			echo '<div class="notice notice-success is-dismissible"><p>Settings have been reset to default.</p></div>';
+			delete_option( 'aspirepress-reset' );
+		}
+	}
+
+	/**
+	 * Get the value of a Setting by giving priority to hard coded values.
+	 *
+	 * @param string $setting_name The name of the settings field.
+	 * @param mixed  $default_value The Default value to return if the field is not defined.
+	 *
+	 * @return string The value of the settings field.
+	 */
+	public function get_setting( $setting_name, $default_value = false ) {
+		if ( null === $this->options ) {
+			$options = get_option( $this->option_name, false );
+			/**
+			 * If the options are not set load defaults.
+			 */
+			if ( false === $options ) {
+				$options = $this->get_default_settings();
+				update_option( $this->option_name, $options );
```

Comment:
is $options here safe? I understand it is either default (presumably safe) or what is from the DB - not necessarily clean data.

Reply:
That comes from a manually defined array in code in $this->get_default_settings()